core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -439,6 +442,28 @@ public void removeInstance(@Nonnull IdentityUser user, Long stackId, String inst
         }
     }
 
+    public void removeInstances(IdentityUser user, Long stackId, Set<String> instanceIds) {
+        Stack stack = get(stackId);
+        authorizationService.hasWritePermission(stack);
+        Map<String, Set<Long>> groupToDiscoveryPrivateIdMap = new HashMap<>();
+        for (String instanceId: instanceIds) {
+            InstanceMetaData metadata = instanceMetaDataRepository.findByInstanceId(stackId, instanceId);
+            if (metadata == null) {
+                throw new NotFoundException(String.format("Metadata for instance %s not found.", instanceId));
+            }
+            if (groupToDiscoveryPrivateIdMap.containsKey(metadata.getInstanceGroupName())) {
+                Set<Long> privateIds = groupToDiscoveryPrivateIdMap.get(metadata.getInstanceGroupName());
+                privateIds.add(metadata.getPrivateId());
+                groupToDiscoveryPrivateIdMap.put(metadata.getInstanceGroupName(), privateIds);
+            } else {
+                Set<Long> privateIds = new LinkedHashSet<>();
+                privateIds.add(metadata.getPrivateId());
+                groupToDiscoveryPrivateIdMap.put(metadata.getInstanceGroupName(), privateIds);
+            }
+        }
+        groupToDiscoveryPrivateIdMap.forEach((k, v) -> flowManager.triggerStackRemoveInstances(stackId, k, v));
```

Comment:
We have a check about the running flows on the stack. I guess this will fail if we trigger more than one flow at a time. If we are 100% sure that those flows are ok in parallel, possible solution would be to delete `Promise` from the `ClusterAndStackDownscaleTriggerEvent` but the best solution is to create a separate flow for removing more than one instance.

Reply:
Yes, we should create a new flow for terminating multiple instances at the same time, because allowing parallel flows would cause headaches I think.